questions/119-medium-replaceall/template.ts

```diff
@@ -1,5 +1,16 @@
+//Similar to the previous excercise, for this one I use ReplaceString recursively to replace all occurrences of the 'From' string with the 'To' string
+type IsEmptyString<S extends string> = S extends '' ? true : false
+
+type ReplaceString<
+  S extends string,
+  From extends string,
+  To extends string
+> = S extends `${infer L}${From}${infer R}`
+  ? `${L}${To}${ReplaceString<R, From, To>}`
```

Comment:
This is a great solution.

What would happen if I do:

```suggestion
  ? `${ReplaceString<R, From, To>}${L}${To}`
```

I know this does not meet the challenge requirement but why?

Reply:
It would still perform the replacement but in the wrong order, the result of the recursive replacement would be placed at the beginning, it would work as a reverse replacement.